### PR TITLE
Split target.bzl further

### PR DESCRIPTION
**Pull request by maxwellE**
Splits up target.bzl further

### CODE REVIEW

File: xcodeproj/internal/search_paths.bzl [hunk 1]
```diff
@@ -0,0 +1,69 @@
+""" Functions for processing search paths """
```
Comment (brentleyjones) [inline xcodeproj/internal/search_paths.bzl:1]:
```suggestion
"""Functions for processing search paths."""
```

File: xcodeproj/internal/search_paths.bzl [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+load(":collections.bzl", "set_if_true")
+load(
+    ":files.bzl",
+    "file_path_to_dto",
+    "parsed_file_path",
+)
+
+def process_search_paths(*, cc_info, objc, opts_search_paths):
+    """ Process search paths
```
Comment (brentleyjones) [inline xcodeproj/internal/search_paths.bzl:11]:
```suggestion
    """Processes search paths.
```

File: xcodeproj/internal/search_paths.bzl [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    Args:
+        cc_info: A CcInfo provider object
+        objc: A ObjC provider object
+        opts_search_paths: optiions for search paths
```
Comment (brentleyjones) [inline xcodeproj/internal/search_paths.bzl:16]:
```suggestion
        cc_info: The `CcInfo` provider for the target.
        objc: The `ObjcProvider` provider for the target.
        opts_search_paths: A value returned from `create_opts_search_paths`.
```

File: xcodeproj/internal/search_paths.bzl [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    Returns:
+        All search paths for the provided objects
```
Comment (brentleyjones) [inline xcodeproj/internal/search_paths.bzl:19]:
```suggestion
        A DTO `dict`.
```

File: xcodeproj/internal/target.bzl [hunk 5]
```diff
@@ -45,299 +39,19 @@ load(
...
-    )
+load(
+    ":top_level_targets.bzl",
+    "process_defines",
+    "process_dependencies",
+    "process_modulemaps",
+    "process_sdk_links",
+    "process_swiftmodules",
+    "process_top_level_properties",
+    "process_top_level_target",
```
Comment (brentleyjones) [inline xcodeproj/internal/target.bzl:53]: Not all of these are top-level only. Some things were improperly grouped under that comment, sorry.

Comment (maxwellE) [reply xcodeproj/internal/target.bzl:53]: Where would you like these to live?

Comment (brentleyjones) [reply xcodeproj/internal/target.bzl:53]: I think these:
```
    "process_defines",
    "process_dependencies",
    "process_modulemaps",
    "process_sdk_links",
    "process_swiftmodules"
```

along with `process_search_paths` all seem to be related, sorta as "target_properties.bzl" or something? They are all for compiling/linking of targets, regardless if they are top level or not.

File: xcodeproj/internal/target.bzl [hunk 6]
```diff
@@ -46,350 +39,24 @@ load(
...
+    "process_modulemaps",
+    "process_sdk_links",
+    "process_swiftmodules",
+)
+load(
+    ":top_level_targets.bzl",
+    "process_top_level_properties",
+    "process_top_level_target",
+    "should_bundle_resources",
+    "should_include_outputs",
```
Comment (brentleyjones) [inline xcodeproj/internal/target.bzl:58]: These two should also be in `target_properties.bzl`, as they are used by library targets as well.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**